src/code/z_actor.c

```diff
@@ -412,103 +423,111 @@ void func_8002C124(TargetContext* targetCtx, PlayState* play) {
-            actorArg = NULL;
+    if ((lockOnActor != NULL) && (targetCtx->rotZTick == 0)) {
+        Actor_ProjectPos(play, &lockOnActor->focus.pos, &projectedFocusPos, &invW);
+        if (((projectedFocusPos.z <= 0.0f) || (1.0f <= fabsf(projectedFocusPos.x * invW))) ||
+            (fabsf(projectedFocusPos.y * invW) >= 1.0f)) {
+            lockOnActor = NULL;
         }
     }
 
-    if (actorArg != NULL) {
-        if (actorArg != targetCtx->targetedActor) {
-            func_8002BE98(targetCtx, actorArg->category, play);
-            targetCtx->targetedActor = actorArg;
+    if (lockOnActor != NULL) {
+        if (lockOnActor != targetCtx->lockOnActor) {
+            // Lock On entries need to be re-initialized when changing the targeted actor
+            Target_InitReticle(targetCtx, lockOnActor->category, play);
+            targetCtx->lockOnActor = lockOnActor;
 
-            if (actorArg->id == ACTOR_EN_BOOM) {
-                targetCtx->unk_48 = 0;
+            if (lockOnActor->id == ACTOR_EN_BOOM) {
+                // Avoid drawing the lock on triangles on the boomerang
+                targetCtx->reticleFadeAlphaControl = 0;
             }
 
-            lockOnSfxId = CHECK_FLAG_ALL(actorArg->flags, ACTOR_FLAG_0 | ACTOR_FLAG_2) ? NA_SE_SY_LOCK_ON
-                                                                                       : NA_SE_SY_LOCK_ON_HUMAN;
+            lockOnSfxId = CHECK_FLAG_ALL(lockOnActor->flags, ACTOR_FLAG_0 | ACTOR_FLAG_2) ? NA_SE_SY_LOCK_ON
+                                                                                          : NA_SE_SY_LOCK_ON_HUMAN;
             Sfx_PlaySfxCentered(lockOnSfxId);
         }
 
-        targetCtx->targetCenterPos.x = actorArg->world.pos.x;
-        targetCtx->targetCenterPos.y = actorArg->world.pos.y - (actorArg->shape.yOffset * actorArg->scale.y);
-        targetCtx->targetCenterPos.z = actorArg->world.pos.z;
+        targetCtx->lockOnPos.x = lockOnActor->world.pos.x;
+        targetCtx->lockOnPos.y = lockOnActor->world.pos.y - (lockOnActor->shape.yOffset * lockOnActor->scale.y);
+        targetCtx->lockOnPos.z = lockOnActor->world.pos.z;
+
+        if (targetCtx->rotZTick == 0) {
+            lockOnStep = (500.0f - targetCtx->reticleRadius) * 3.0f;
+            lockOnStep = CLAMP(lockOnStep, 30.0f, 100.0f);
 
-        if (targetCtx->unk_4B == 0) {
-            temp5 = (500.0f - targetCtx->unk_44) * 3.0f;
-            temp6 = (temp5 < 30.0f) ? 30.0f : ((100.0f < temp5) ? 100.0f : temp5);
-            if (Math_StepToF(&targetCtx->unk_44, 80.0f, temp6) != 0) {
-                targetCtx->unk_4B++;
+            if (Math_StepToF(&targetCtx->reticleRadius, 80.0f, lockOnStep)) {
+                targetCtx->rotZTick++;
             }
         } else {
-            targetCtx->unk_4B = (targetCtx->unk_4B + 3) | 0x80;
-            targetCtx->unk_44 = 120.0f;
+            // 0x80 is or'd to avoid getting this value be set to zero
```

Comment:
"0x80 is or'd to avoid a value of zero", or something. as is the sentence doesnt read right